File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string temp = "";
        for(int i = 0; i<strs[0].size(); i++){
            for(int j = 1; j<strs.size(); j++){
                if(i == strs[j].size() || strs[0][i] != strs[j][i]) return temp;
            }
            temp += strs[0][i];
        }
    return temp;
    }
};
// time O(m*n);
// space O(1)

File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let ans = "";
    for(let i = 0; i<strs[0].length; i++){
        for(let j = 1; j<strs.length; j++){
            if(i == strs[j].length || strs[j][i] != strs[0][i]) return ans;
        }
        ans += strs[0][i];
    }
    return ans;
};